lib/rucio/tests/test_rule.py

```diff
@@ -1016,6 +1018,108 @@ def test_rule_on_archive(self):
+            for i in range(8, 11):
+                add_replica(rse_id=get_rse_id('MOCK3'), scope=scope, name='file_%06d.data' % i, bytes=10000 + i, account=account)
+            for i in range(11, 17):
+                add_replica(rse_id=get_rse_id('MOCK4'), scope=scope, name='file_%06d.data' % i, bytes=10000 + i, account=account)
+            for i in range(17, 21):
+                add_replica(rse_id=get_rse_id('MOCK5'), scope=scope, name='file_%06d.data' % i, bytes=10000 + i, account=account)
+            for i in range(21, 25):
+                add_replica(rse_id=get_rse_id('MOCK'), scope=scope, name='file_%06d.data' % i, bytes=10000 + i, account=account)
+            for i in range(25, 26):
+                add_replica(rse_id=get_rse_id('MOCK5'), scope=scope, name='file_%06d.data' % i, bytes=10000 + i, account=account)
+
+            add_did(scope=scope, name='ds1', type='DATASET', account=account)
+            attach_dids(scope=scope, name='ds1', dids=[{'scope': scope, 'name': 'file_%06d.data' % i} for i in range(1, 10 + 1)], account=account)
+            add_did(scope=scope, name='ds2', type='DATASET', account=account)
+            attach_dids(scope=scope, name='ds2', dids=[{'scope': scope, 'name': 'file_%06d.data' % i} for i in range(11, 20 + 1)], account=account)
+            add_did(scope=scope, name='ds3', type='DATASET', account=account)
+            attach_dids(scope=scope, name='ds3', dids=[{'scope': scope, 'name': 'file_%06d.data' % i} for i in (range(21, 25 + 1) + [1, 2, 11, 12])], account=account)
+            add_did(scope=scope, name='container12', type='CONTAINER', account=account)
+            attach_dids(scope=scope, name='container12', dids=[{'scope': scope, 'name': 'ds1'}, {'scope': scope, 'name': 'ds2'}, ], account=account)
+            add_did(scope=scope, name='container13', type='CONTAINER', account=account)
+            attach_dids(scope=scope, name='container13', dids=[{'scope': scope, 'name': 'ds1'}, {'scope': scope, 'name': 'ds3'}, ], account=account)
+            add_did(scope=scope, name='container1213', type='CONTAINER', account=account)
+            attach_dids(scope=scope, name='container1213', dids=[{'scope': scope, 'name': 'container12'}, {'scope': scope, 'name': 'container13'}, ], account=account)
+
+        account = self.jdoe
+
+        # test1 : ALL grouping -> select MOCK for all 3 datasets
+        scope = InternalScope(('scope1_' + str(uuid()))[:24])  # scope field has max 25 chars
+        add_scope(scope, account)
+        mktree(scope, account)
+        rule_ids = add_rule(dids=[{'scope': scope, 'name': 'container1213'}], copies=1, rse_expression='MOCK|MOCK3|MOCK4|MOCK5', grouping='ALL',
+                            account=account, weight=None, lifetime=None, locked=False, subscription_id=None)
+        rule = get_rule(rule_ids[0])
+        print(rule['locks_ok_cnt'], rule['locks_replicating_cnt'])
```

Comment:
Please remove the print statements, the CI will be very spammy otherwise. 😄

Reply:
where exactly is the CI spammed ?
if the test passes, neither stdout nor logging output is printed

Reply:
You are right, ignore my comment ;-)